lib/ansible/modules/cloud/vmware/vmware_drs_group.py

```diff
@@ -0,0 +1,600 @@
+        self.__result = dict()
+        self.__changed = False
+        self.__state = state
+
+        if datacenter_name is not None:
+
+            self.__datacenter_obj = find_datacenter_by_name(self.content, self.__datacenter_name)
+
+            if self.__datacenter_obj is None and module.check_mode is False:
+                raise Exception("Datacenter '%s' not found" % self.__datacenter_name)
+
+        self.__cluster_obj = find_cluster_by_name(content=self.content,
+                                                  cluster_name=self.__cluster_name,
+                                                  datacenter=self.__datacenter_obj)
+
+        # Throw error if cluster does not exist
+        if self.__cluster_obj is None and module.check_mode is False:
+            raise Exception("Cluster '%s' not found" % self.__cluster_name)
+
+        # Dont populate lists if we are deleting group
+        if state == 'present':
+
+            if self.__vm_list is not None:
+                self.__set_vm_obj_list(vm_list=self.__vm_list)
+
+            if self.__host_list is not None:
+                self.__set_host_obj_list(host_list=self.__host_list)
+
+    def get_msg(self):
+        """
+        Returns message for Ansible result
+        Args: none
+
+        Returns: string
+        """
+        return self.__msg
+
+    def get_result(self):
+        """
+        Returns result for Ansible
+        Args: none
+
+        Returns: dict
+        """
+        return self.__result
+
+    def __set_result(self, group_obj):
+        """
+        Creates result for successfull run
+        Args:
+            group_obj: group pbject
+
+        Returns: None
+
+        """
+        self.__result = dict()
+
+        if (self.__cluster_obj is not None) and (group_obj is not None):
+            self.__result[self.__cluster_obj.name] = []
+            self.__result[self.__cluster_obj.name].append(self.__normalize_group_data(group_obj))
+
+    def get_changed(self):
+        """
+        Returns if anything changed
+        Args: none
+
+        Returns: boolean
+        """
+        return self.__changed
+
+    def __set_vm_obj_list(self, vm_list=None, cluster_obj=None):
+        """
+        Function pupulate vm object list from list of vms
+        Args:
+            vm_list: List of vm names
+
+        Returns: None
+
+        """
+
+        if vm_list is None:
+            vm_list = self.__vm_list
+
+        if cluster_obj is None:
+            cluster_obj = self.__cluster_obj
+
+        if vm_list is not None:
+
+            for vm in vm_list:
+
+                if self.module.check_mode is False:
+
+                    # Get host data
+                    vm_obj = find_vm_by_id(content=self.content, vm_id=vm,
+                                           vm_id_type='vm_name', cluster=cluster_obj)
+
+                    if vm_obj is None:
+                        raise Exception("VM %s does not exist in cluster %s" % (vm,
```

Comment:
Same as above, please fail the module, instead of raising an exception
```suggestion
            self.module.fail_json(msg="VM %s does not exist in cluster %s" % (vm,
```